api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
Its true that we can update voided obs and cannot track such a change. To track such a change maybe we could change isDirty() property of the obs to false on voiding it and change it to true if a voided obs is modified. Then we could keep a check that if an obs is voided and is marked dirty means than it means has been modified.  Anyway I do think we should validate an obs obs before unvoiding it. The fact whether an obs has been modified after voiding or not should not affect that.(the current test still seem to pass) This is what I understand. What do you think @wluyima @dkayiwa :)